src/devices/QwiicButton/QwiicButton.InterruptConfiguration.cs

```diff
@@ -0,0 +1,92 @@
+﻿//// Licensed to the .NET Foundation under one or more agreements.
+//// The .NET Foundation licenses this file to you under the MIT license.
+//// See the LICENSE file in the project root for more information.
+
+using Iot.Device.QwiicButton.RegisterMapping;
+
+namespace Iot.Device.QwiicButton
+{
+    public sealed partial class QwiicButton
```

Comment:
unless this class joined together is couple thousands line long I prefer this be in a single file

Reply:
I'm a big fan of separation of concerns and keeping classes small. Since this conflicts with the wish of having all functionality on one device class, using partial classes is in my opinion a decent compromise.